src/main/scala/mesosphere/marathon/api/WebJarServlet.scala

```diff
@@ -0,0 +1,67 @@
+package mesosphere.marathon.api
+
+import javax.servlet.http.{ HttpServlet, HttpServletRequest, HttpServletResponse }
+
+import mesosphere.marathon.io.IO
+import org.apache.log4j.Logger
+
+class WebJarServlet extends HttpServlet {
+
+  private[this] val log = Logger.getLogger(getClass)
+
+  override def doGet(req: HttpServletRequest, resp: HttpServletResponse): Unit = {
+    //extract request data
+    val jar = req.getServletPath // e.g. /ui
+    var resource = req.getPathInfo // e.g. /fonts/icon.gif
+    if (resource.endsWith("/")) resource = resource + "index.html" // welcome file
+    val file = resource.split("/").last //e.g. icon.gif
+    val mediaType = file.split("\\.").lastOption.getOrElse("") //e.g. gif
+    val mime = Option(getServletContext.getMimeType(file)).getOrElse(mimeType(mediaType)) //e.g plain/text
+    val resourceURI = s"/META-INF/resources/webjars$jar$resource"
+
+    //log request data, since the names are not very intuitive
+    if (log.isDebugEnabled) {
+      log.debug(
+        s"""
+         |pathinfo: ${req.getPathInfo}
+         |context: ${req.getContextPath}
+         |servlet: ${req.getServletPath}
+         |path: ${req.getPathTranslated}
+         |uri: ${req.getRequestURI}
+         |jar: $jar
+         |resource: $resource
+         |file: $file
+         |mime: $mime
+         |resourceURI: $resourceURI
+       """.stripMargin)
+    }
+
+    def sendResource(): Unit = {
```

Comment:
If you use nested methods, I'd prefer if you put them first. So you have the following structure:

* methods that are only used in this method and help with the implementation
* code that uses the methods and "wires" them together.

I do like it if the methods do not simply close over local variables but have them as arguments.